backend/useragent/user_agent.go

```diff
@@ -0,0 +1,62 @@
+package useragent
+
+import (
+	"errors"
+	"regexp"
+)
+
+var (
+	userAgentRegex   = regexp.MustCompile(`^Grafana/([0-9]+\.[0-9]+\.[0-9]+(?:-[a-zA-Z0-9]+)?) \(([a-zA-Z0-9]+); ([a-zA-Z0-9]+)\)$`)
+	errInvalidFormat = errors.New("invalid user agent format")
+)
+
+// UserAgent represents a Grafana user agent.
+// Its format is "Grafana/<version> (<os>; <arch>)"
+// Example: "Grafana/7.0.0-beta1 (darwin; amd64)", "Grafana/10.0.0 (windows; x86)"
+type UserAgent struct {
+	grafanaVersion string
+	arch           string
+	os             string
+}
+
+// New creates a new UserAgent.
+// The version must be a valid semver string, and the os and arch must be valid strings.
+func New(grafanaVersion, os, arch string) (*UserAgent, error) {
+	ua := &UserAgent{
+		grafanaVersion: grafanaVersion,
+		os:             os,
+		arch:           arch,
+	}
+
+	return NewFromString(ua.String())
+}
+
+// NewFromString creates a new UserAgent from a string.
+func NewFromString(s string) (*UserAgent, error) {
+	matches := userAgentRegex.FindStringSubmatch(s)
+	if len(matches) != 4 {
+		return nil, errInvalidFormat
+	}
+
+	return &UserAgent{
+		grafanaVersion: matches[1],
+		os:             matches[2],
+		arch:           matches[3],
+	}, nil
+}
+
+func (ua *UserAgent) GrafanaVersion() string {
+	return ua.grafanaVersion
+}
+
+func (ua *UserAgent) OS() string {
+	return ua.os
+}
+
+func (ua *UserAgent) Arch() string {
+	return ua.arch
+}
```

Comment:
Is this needed? Feels a bit complicated compared with plugin version.

Reply:
For now actually not. We could actually choose to not expose anything (including Grafana Version) if we want to deprecate usage of Grafana version information so that it should only be used in the context of user agent info (for logging, tracing, etc.)

Reply:
Okay sounds like maybe wait with exposing it then?

Reply:
Will do 👍